src/components/FilterBar/FilterDisplay/Inputs/RelativeDateRangeInput.react.js

```diff
@@ -0,0 +1,208 @@
+  }
+
+  relativeOption(optionKey) {
+    var optionItem = this.props.relativeOptions[optionKey];
+    return <option
+      key={optionKey}
+      value={optionItem.value !== undefined ? optionItem.value : optionKey}
+      data-from={optionItem.from} data-to={optionItem.to}>
+      {optionKey}
+    </option>
+  }
+
+  showDateInputs() {
+    return (
+      <div>
+        <DateInput value={this.state.value} filterUid={this.props.filterUid} displayFrom={this.state.displayFrom} displayTo={this.state.displayTo} />
+      </div>
+    )
+  }
+
+  showRelativeRangeInputs() {
+    return (
+      <div>
+        <div className="input-group">
+          <span className="input-group-addon">
+            Today
+          </span>
+          <input
+            type="number"
+            className="form-control"
+            onBlur={this.onBlur.bind(this)}
+            onChange={(e) => this.onChangeFrom(e)}
+            placeholder="+/- days"
+            value={this.state.value.from}
+          />
+          <span className="input-group-addon">
+            day(s)
+          </span>
+          <span className="input-group-addon">
+            {this.momentFormatted(this.state.value.from)}
+          </span>
+        </div>
+        <div className="input-group">
+          <span className="input-group-addon">
+            Today
+          </span>
+          <input
+            type="number"
+            className="form-control"
+            onBlur={this.onBlur.bind(this)}
+            onChange={(e) => this.onChangeTo(e)}
+            placeholder="+/- days"
+            value={this.state.value.to}
+          />
+          <span className="input-group-addon">
+            day(s)
+          </span>
+          <span className="input-group-addon">
+            {this.momentFormatted(this.state.value.to)}
+          </span>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    return (
+      <div className="col-sm-10">
+        <select className="form-control"
+          onChange={this.onRelativeChange.bind(this)}
+          value={this.state.value.value}
+        >
+          {Object.keys(this.props.relativeOptions).map((optionKey) => (
+            this.relativeOption(optionKey)
+          ))}
+        </select>
+
+        {
+          this.state.operator == "absolute" ? this.showDateInputs() : this.showRelativeRangeInputs()
+        }
+
+        <div className="input-group">
+          <label className="radio-inline">
+            <input
+              type="radio"
+              name="operator"
+              value="absolute"
+              checked={this.isAbsolute()}
+              onChange={(e) => this.setState({ operator: e.target.value })}
+            />
+            Absolute
+          </label>
+
+          <label className="radio-inline">
+            <input
+              type="radio"
+              name="operator"
+              value="relative"
+              checked={this.state.operator == "relative"}
```

Comment:
Seems a little strange that you have the `this.isAbsolute()` helper method, but not one for relative.
Is there any way where we get ourselves into a bad state and end up with neither checked?
I'm wondering if it would be better to do `checked={!isAbsolute()}` to ensure we never hit that.

Reply:
Sure, I must have forgotten.